packages/vuetify/src/components/VDialog/VDialog.tsx

```diff
@@ -0,0 +1,124 @@
+// Styles
+import './VDialog.sass'
+
+import { defineComponent, mergeProps, ref, watch } from 'vue'
+import { VOverlay } from '@/components/VOverlay'
+
+// Helpers
+// import {
+//   convertToUnit,
+//   keyCodes,
+// } from '../../util/helpers'
+import { makeProps } from '@/util/makeProps'
+import { makeDimensionProps, useDimension } from '@/composables/dimensions'
+import { makeTransitionProps } from '@/composables/transition'
+import { useProxiedModel } from '@/composables/proxiedModel'
+import { VDialogTransition } from '@/components/transitions'
+
+export default defineComponent({
+  name: 'VDialog',
+
+  props: makeProps({
+    fullscreen: Boolean,
+    origin: {
+      type: String,
+      default: 'center center',
+    },
+    retainFocus: {
+      type: Boolean,
+      default: true,
+    },
+    scrollable: Boolean,
+    modelValue: Boolean,
+    ...makeDimensionProps({ width: 'auto' }),
+    ...makeTransitionProps({
+      transition: { component: VDialogTransition },
+    }),
+  }),
+
```

Comment:
declare emits

Reply:
Should I declare `click:outside` here too even though it isn't actually emitted by this component?

Reply:
No that shouldn't be necessary